src/compiler/checker.ts

```diff
@@ -38315,17 +38320,20 @@ namespace ts {
         function checkTryStatement(node: TryStatement) {
             // Grammar checking
             checkGrammarStatementInAmbientContext(node);
-
             checkBlock(node.tryBlock);
+
             const catchClause = node.catchClause;
             if (catchClause) {
                 // Grammar checking
                 if (catchClause.variableDeclaration) {
                     const declaration = catchClause.variableDeclaration;
+                    if (isBindingPattern(declaration.name)) {
+                        forEach(declaration.name.elements, checkSourceElement);
+                    }
                     const typeNode = getEffectiveTypeAnnotationNode(getRootDeclaration(declaration));
                     if (typeNode) {
                         const type = getTypeForVariableLikeDeclaration(declaration, /*includeOptionality*/ false);
-                        if (type && !(type.flags & TypeFlags.AnyOrUnknown)) {
+                        if (type === errorType) {

```

Comment:
I think this should be `isErrorType` (looking at the rest of the code).

Reply:
However, I'm a bit confused how this works after the change; before this read as "if it's not any or unknown, error", which is the spirit of the diagnostic.